src/gnn_reco/models/task/reconstruction.py

```diff
@@ -75,3 +75,19 @@ def _forward(self, x):
         log_var = x[:,1]
         pred = torch.stack((energy, log_var), dim=1)
         return pred
+
+class VertexReconstruction(Task):
+    # Requires four features, x, y, z and t
+    nb_inputs = 4
+
+    def _forward(self, x):
+        # Initally do nothing to the output
+        return x.unsqueeze(1) 
```

Comment:
The input `x` to `_forward` should be of shape [N, 4], I believe. By unsqueezing, I suspect the output ends up having shape [N, 1, 4]. Is that the case? And is that what you want to happen. I would expect that just returning `x` would suffice, but I might be wrong.

Reply:
Perhaps training would be simpler if the components in `x` were scaled up according to the rough physical dimensions of the IceCube detector; e.g. `x[:,:3] = 100. * x[:,:3]` as initially the elements in `x` will of order 1. Have you had a chance to train with this task? How have you handled this scale?

Reply:
The unsqueeze is just copied over from the other simple tasks like `AngularReconstruction` and `EnergyReconstruction`. It works, but only because I in the loss function do: `diff = prediction[:,0] - target`  which is done in the current `LogCoshLoss `function. Ive tested removing the unsqueezing and the `[:,0] `, and everything still works as before, so I will implement that

Reply:
Right now, I only scale the target in the loss function (and only for time). The model is able to output values in the correct xyz range by it's own, but it will probably make it easier to give it a bit of help by scaling by rough orders of magnitude in the task, so I will do that as well!